### DISCUSSION TITLE
Integrations for cloud file storage (google drive, one drive, s3, etc.)

### DISCUSSION THREAD

**Discussion by mdvertola** [open-webui/open-webui#4079] [General]
**Is your feature request related to a problem? Please describe.**
It would be super helpful if we could import large amounts of documents into environments by integrations with things like google drive, sharepoint, onedrive, etc. otherwise if we would like to import and embed thousands of documents, uploading becomes cumbersome. It also poses the issues of keeping documents up to date as well.

**Describe the solution you'd like**
Within the `/workspace/documents/` or the `/admin/settings/` (side nav documents) create the ability to integrate with a cloud file storage provider.

**Describe alternatives you've considered**
Manual mass file upload, or cron job (run outside of open-webui) that will pull files from cloud file storage, then drop the files in DOCS_DIR (/data/docs).

**Additional context**
I think this feature would be super helpful when considering this from the use case of small/medium businesses that are looking to incorporate their files into the platform for Q&A/summarization related tasks.

**Comment by silentoplayz** [comment]
Related https://github.com/open-webui/open-webui/issues/1293

**Comment by danielrosehill** [comment]
I'd love this but also the ability (more important to me) to go in the other direction - save a select output to your Gdrive/ wherever for later reference. Working on a tool for doing that (selective saving from a chat)

**Comment by hades200082** [comment]
Under Admin Panel > Settings > Documents there are toggles for OneDrive and Google Drive ... what do they do? I can't find any mention of them in the docs, or anywhere else in the UI after turning them on.

**Reply by AntoineMaille** [reply]
Same here

**Reply by silentoplayz** [reply]
Related PRs https://github.com/open-webui/open-webui/pull/7900 and https://github.com/open-webui/open-webui/pull/10690 since documentation is either limited or isn't written concerning these two integrations.

**Reply by gizbo** [reply]
Looks like a work in progress. After enabling the OneDrive toggle, I then got the following option:
![Screenshot 2025-04-03 160500](https://github.com/user-attachments/assets/42eb01ad-c22d-4ebb-8a53-2fa3db6eea82)
Nothing doing anything.

**Reply by richjdsmith** [reply]
Anything ever come of this?

**Comment by adityapandey216** [comment]
I think these are config driven Google drive API key, secret needs to be set in the environment variables

**Reply by TrilceAC** [reply]
Indeed:

- Google Drive Integration: https://docs.openwebui.com/features/rag/#google-drive-integration
- Google Drive env vars for integration with RAG: https://docs.openwebui.com/getting-started/env-configuration/#google-drive
- OneDrive env vars for integration with RAG: https://docs.openwebui.com/getting-started/env-configuration/#onedrive

Got this information from this reddit answer in Spanish: https://www.reddit.com/r/OpenWebUI/comments/1kfeyn0/comment/mr8o11z/

**Comment by brtptrs** [comment]
Nextcloud (WebDav) integration would be interesting as well.

**Comment by hdnh2006** [comment]
I have made a tutorial about how to integrate Google Drive into Open WebUI

- Here you have the free medium article: https://henrynavarro.org/how-to-integrate-google-drive-with-open-webui-e525c8f3f82e?sk=a9f48cd1590e3e8dca052f7fe9f12765
- And for those spanish speakers, you have a tutorial in my channel: https://youtu.be/cGBjAned72A?si=99TRUKgHZQXWcNs2
